Code hygiene

towhans opened the pull request:
Changes motivated by code review

Commit:
Refactor out duplicate code

Commit:
Use nil instead of :unknown atom

- to represent situation when high watermark offset is unknown to consumer
  (prior to receiving first kafka_message_set)

Commit:
Bump version

liskin commented on lib/kafka_gen_stage/consumer_logic.ex:
```diff
@@ -83,7 +67,7 @@ defmodule KafkaGenStage.ConsumerLogic do
         ) :: dispatch()
   defp prepare_dispatch(queue, 0, to_send, to_ack, _high_wm) do
-    {to_send, to_ack, 0, queue}
+    {to_send, to_ack, queue}
```
Now that we've simplified the function it becomes obvious it's not really a `prepare_dispatch`, all it does it dequeue as many msgs as possible up to `demand` from the queue. The `high_wm` parameter isn't used at all. `to_ack` is simply the offset of the last message dequeued, which conveniently happens to be the first in the returned list, thus cheap to read.

Therefore I suggest:

* turn this into a `dequeue` function that takes up to `demand` msgs from `queue`
* in `prepare_dispatch/4` calculate to_ack as `case to_send do [] -> :no_ack, [{offset, …}|_] -> offset end`
* drop the `dispatch()` type
* rename `return_dispatch()` to `dispatch()`

towhans replied:
I thought about `dequeue` too. I will do it.

liskin commented on lib/kafka_gen_stage/consumer.ex:
```diff
@@ -141,7 +141,7 @@ defmodule KafkaGenStage.Consumer do
           consumer_ref: reference(),
           queue: :queue.queue(),
-          high_wm_offset: non_neg_integer() | atom(),
+          high_wm_offset: non_neg_integer(),
```
I think `nil` still needs to be mentioned here. And in the type of `prepare_dispatch/4` as well.

Commit:
Correctly detect end of stream

- when topic is empty
- when starting to read at the very end
- API change: bulk_transformer has second parameter is_end_of_stream

Commit:
Bump version to 2.0.2
